cmd/goordinator/main.go

```diff
@@ -318,25 +379,53 @@ func main() {
 		zap.String("github_api_token", hide(config.GithubAPIToken)),
 		zap.String("log_format", config.LogFormat),
 		zap.String("log_time_key", config.LogTimeKey),
+		zap.Bool("autoupdater.trigger_on_auto_merge", config.Autoupdater.TriggerOnAutoMerge),
+		zap.Strings("autoupdater.labels", config.Autoupdater.Labels),
+		zap.Any("autoupdater.repositories", config.Autoupdater.Repositories),
+		zap.String("autoupdater.http_endpoint", config.Autoupdater.Endpoint),
 		zap.String("rules", rules.String()),
 	)
 
 	goodbye.Register(func(_ context.Context, sig os.Signal) {
 		logger.Info(fmt.Sprintf("terminating, received signal %s", sig.String()))
 	})
-	evLoopchan := make(chan *github.Event, EventChannelBufferSize)
+
+	var chans []chan<- *github.Event
+
+	mux := http.NewServeMux()
+
+	autoupdater, ch := startPullRequestAutoupdater(config, githubClient, mux)
+	if ch != nil {
+		chans = append(chans, ch)
+	}
+
+	if len(rules) > 0 {
+		evLoopchan := make(chan *github.Event, EventChannelBufferSize)
+		chans = append(chans, evLoopchan)
+
+		evLoop := goordinator.NewEventLoop(
+			evLoopchan,
+			rules,
+			goordinator.WithActionRoutineDeferFunc(panicHandler),
+		)
+
+		go evLoop.Start() // todo: implement ordered shutdown, wait for termination
```

Comment:
is this critical? what's the behavior on shutdown atm?

Reply:
It terminates abruptly when receiving the signal.
There are currently not operations that must finish otherwise something is an unrecoverable state.

When goordinator is started again, the state of the autoupdater will be synced by querying the github API.
Actions triggered because of github webhook events that were aborted abruptly will be lost, that is also the same when goordinator is terminated.

It does not make a real difference currently